mh/session.py

```diff
@@ -315,10 +315,9 @@ def get_layer_host_data(self):
     def get_optional_fields(self):
         """LayerUserInfo's optional fields."""
         location = int(self.is_in_quest())  # City - 0, Quest - 1
-        hunter_rank, = struct.unpack_from(">H", self.hunter_info.data, 0)
-        weapon_icon, = struct.unpack_from(">B", self.hunter_info.data, 0x10)
-        weapon_icon -= 7  # Skip armor pieces and start at index zero
```

Comment:
Won't removing this `-7` part break the LayerUserInfo?

Reply:
(I tested it)

No, because this is no the weapon type ID from the first equipment slot. This is another field in the ‘hunter_info’ that’s send specifically for this purpose. If we were to use the first equipment’s weapon type ID we would have to calculate bowguns (light/medium/heavy)